gammapy/modeling/models/spectral.py

```diff
@@ -213,6 +213,31 @@ def evaluate_error(self, energy, epsilon=1e-4):
         """
         return self._propagate_error(epsilon=epsilon, fct=self, energy=energy)
 
+    def pivot_energy(self, energy):
+        """The decorrelation energy for a given spectral model calculated numerically.
+
+        Parameters
+        ----------
+        energy : `~astropy.units.Quantity`
```

Comment:
should there be a default choice of energy?

Reply:
A default energy array could be a good idea. 
Or the other option is for the user to define the `energy_bounds` instead, then there will be no issue with bins.

I believe there was some discussion recently about the validity of certain energy ranges for different functions. In this PR, I had to ensure that we did not have any negative fluxes `y_lo[y_lo > 0]`. Negative values for y_lo are possible if the energy array chosen for this function is not valid for the specific dataset used. For example, using a 'nonsensical' energy range for the Fermi data.